docs/tutorial/from-zero-to-hero-write-your-first-kubernetes-charm/create-a-minimal-kubernetes-charm.md

```diff
@@ -417,6 +417,304 @@ In the output you should see the definition for both containers. You'll be able
+```
+
+You should get an output similar to the one below:
+
+```text                                             
+unit: install_deps> python -I -m pip install 'coverage[toml]' 'ops[testing]' pytest -r /home/ubuntu/juju-sdk-tutorial-k8s/requirements.txt
+unit: commands[0]> coverage run --source=/home/ubuntu/juju-sdk-tutorial-k8s/src -m pytest --tb native -v -s /home/ubuntu/juju-sdk-tutorial-k8s/tests/unit
+=================================================================== test session starts ===================================================================
+platform linux -- Python 3.13.2, pytest-8.3.5, pluggy-1.5.0 -- /home/ubuntu/juju-sdk-tutorial-k8s/.tox/unit/bin/python
+cachedir: .tox/unit/.pytest_cache
+rootdir: /home/ubuntu/juju-sdk-tutorial-k8s
+configfile: pyproject.toml
+collected 1 item
+
+tests/unit/test_charm.py::test_pebble_layer PASSED
+
+==================================================================== 1 passed in 0.11s ====================================================================
+unit: commands[1]> coverage report
+Name           Stmts   Miss Branch BrPart  Cover   Missing
+----------------------------------------------------------
+src/charm.py      18      0      0      0   100%
+----------------------------------------------------------
+TOTAL             18      0      0      0   100%
+  unit: OK (2.12=setup[1.82]+cmd[0.26,0.04] seconds)
+  congratulations :) (2.14 seconds)
+```
+
+Congratulations, you have written your first unit test!
+
+
+(write-integration-tests-for-your-charm)=
+## Write integration tests for your charm
+
+A charm should function correctly not just in a mocked environment but also in a real deployment.
+
+For example, it should be able to pack, deploy, and integrate without throwing exceptions or getting stuck in a `waiting` or a `blocked` status -- that is, it should correctly reach a status of `active` or `idle`.
+
+You can ensure this by writing integration tests for your charm. In the charming world, these are usually written with the [`pytest-operator`](https://github.com/charmed-kubernetes/pytest-operator) library.
+
+In this chapter you will write a small integration test to check that the charm packs and deploys correctly.
+
+### Prepare your test environment
+
+In your `tox.ini` file, add the following new environment:
+
+```
+[testenv:integration]
+description = Run integration tests
+deps =
+    pytest
+    juju
+    pytest-operator
+    -r {tox_root}/requirements.txt
+commands =
+    pytest -v \
+           -s \
+           --tb native \
+           --log-cli-level=INFO \
+           {posargs} \
+           {[vars]tests_path}/integration
+```
+
+If you used `charmcraft init --profile kubernetes` in previous steps to bootstrap your project, the `testenv:integration` section is already in the `tox.ini` config file.
+
+### Prepare your test directory
+
+Create  a  `tests/integration` directory:
+
+```text
+mkdir ~/fastapi-demo/tests/integration
+```
```

Comment:
I think for consistency we should do what you did above with "In your project root...". This is better in some ways, since you can run it anywhere, but I think for a beginner being told where to start is actually easier.

Reply:
For consistency, I'll use the "project root" approach - When following other tutorials, I sometimes find that I don't want or can't use what the tutorial proposed as a working directory, so even if it's `mkdir -p ~/proejct/root/some/path/`, I can't always copy-paste. For example maybe the users don't want to use their home directory as a working directory.